ims/einstein/iscsi/tgt.py

```diff
@@ -11,180 +12,154 @@
 
 
 class TGT(ISCSI):
-    '''
-        Class for implementing TGT
-    '''
+    """ Class for implementing TGT """
 
-    # We should get all the config related to ceph in here.
-    # Also, root_password is something that can be avoided.
-
-    # Also removed sudo everywhere since it was causing issues
-    def __init__(self, fs_config_loc, fs_user, root_password):
-        self.arglist = ["service", "tgtd"]
+    def __init__(self, fs_config_loc, fs_user, fs_pool):
         self.TGT_ISCSI_CONFIG = "/etc/tgt/conf.d/"
         self.fs_config_loc = fs_config_loc
         self.fs_user = fs_user
-        self.root_password = root_password
+        self.fs_pool = fs_pool
 
     @log
     def start_server(self):
-        '''
-        Have to parse the output and send a status code.
-        :return:
-        '''
-        # arglist = self.arglist
-        # arglist.append("start")
-        # Need to fix shell=True everywhere
-        command = "service tgtd start"
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        output, err = proc.communicate(self.root_password +
-                                       "\n")
-        logger.debug("Output = %s, Error = %s", output, err)
-        if self.show_status() is 'Running':
+        """
+        Starts the tgt service
+
+        :return: None
+        """
+        try:
+            command = "service tgtd start"
```

Comment:
@ravisantoshgudimetla @chemistry-sourabh Alright, since there's only really one server you need to start, it's ok like this. But, once you have more drivers/services to start, judge the amount of code you can reuse and refactor it in a common module.